map.js

```diff
@@ -11,70 +17,86 @@ let mapY = 0;
+let icon_Cave;
+let icon_Fort;
+let icon_Gate;
+let icon_Inn;
+let icon_Settlement;
+let icon_Mine; //still needs image
+let icon_Landmark; //still needs image
+let icon_Shrine; //still needs image
+
+let locArr;
+
+let debug = false; //makes iframe and guide small by default for map function testing.
 
 function initMap(){
+    //load map cord data
+    fetch("data/locations.json").then(response => response.json()).then(response => locArr = response.elements);
+
     canvas = document.getElementById("canvas_Map");
     mapCanvasContext = canvas.getContext("2d");
     
-    img = document.createElement("img");
-    img.width = 3544;
-    img.height = 2895;
-    img.src = "images/Cyrodil_Upscaled.png";
-    img.onload = drawMap();
-
+    initImgs();
+        
     //center map on imp city
-    mapX = img.width/2.1;
-    mapY = img.height/3.2;
+    mapX = img_Map.width/2.1;
+    mapY = img_Map.height/3.2;
 
+    //Input listeners
     wrapper = document.getElementById("wrapper_Map");
     wrapper.onmousedown = function(){mousedown = true;};
     wrapper.onmouseup = function(){mousedown = false;};
     wrapper.onmouseout = function(){mousedown = false;};
-    wrapper.onmousemove = function(e){
-        if(mousedown){
-            moveMap(e);
-        }
-    };
+    wrapper.onmousemove = function(e){if(mousedown){moveMap(e);}};
     wrapper.onwheel = function(e){zoomMap(e)};
 
     //attaches width to width of iframe
     window.addEventListener("mousemove", iFrameCheck);
 }
- 
-//attaches width to width of iframe
-function iFrameCheck(){
-    if(document.getElementById("iframeContainer")){
-        document.getElementById("iframeContainer").onclick = resizeMap;
-        window.removeEventListener("mousemove", iFrameCheck); //trim listener, it's served it's purpose.
-        resizeMap();
-    }
-}
-
-//matches width of map to Iframe width
-function resizeMap(){
-    if(document.getElementById("iframeContainer")){
-        var ifc = document.getElementById("iframeContainer");
-
-        if(debug){
-            ifc.style.width = "1000px";
-            ifc.style.height = "25px";
-            wrapper.style.height = "580px";
-        }
-
-        wrapper.style.width = ifc.clientWidth + "px";
-        wrapper.style.top = (ifc.clientHeight + 48).toString() + "px";
-    }
-    drawMap();    
-}
 
 function drawMap(){
     //maybe useful for placing icons.
     var aspectw = 443;
     var aspecth = 362;
 
-    mapCanvasContext.drawImage(img, mapX, mapY, (img.width * zoomLevel), (img.height * zoomLevel), 
-                                    0, 0, img.width, img.height);
+    mapCanvasContext.drawImage(img_Map, mapX, mapY, (img_Map.width * zoomLevel), (img_Map.height * zoomLevel), 
+                                    0, 0, img_Map.width, img_Map.height);
+    //draw all map icons //TODO: add in overlay options
+    for(let i = 0; i < locArr.length;i++){
+        if(locArr[i].icon == "Ayleid") drawIcon(icon_Ayleid, locArr[i].approxX, locArr[i].approxY);
```

Comment:
use `switch` statement

Reply:
actually this should be rfactored into a draw(icon, obj) method

Reply:
i did refactor and put this into a switch statement.